dish_grpc_mqtt.py

```diff
@@ -155,10 +193,11 @@ def main():
                 time.sleep(next_loop - now)
             else:
                 break
+    except Terminated:
+        pass
     finally:
         gstate.shutdown()
-
-    sys.exit(rc)
+        sys.exit(rc)
```

Comment:
I'm not sure I like this. Even if it is effectively equivalent logic, it's inconsistent with how this is coded in the other scripts.

Reply:
Actually, now that I've tested this a bit, I can see it's not exactly equivalent. Doing this will prevent the stack trace from being printed. That's not a change I would want.

If the intention is to prevent the KeyboardInterrupt stack trace, then that should just be explicitly caught along with Terminated. I didn't do that myself because it is very useful to know where the code is if it ever gets stuck in an infinite loop that needs to be interrupted, but I can understand if people find it annoying.

Reply:
Yes, it was to stop the exception stack trace from being printed for a SIGINT which it receives on a systemd stop.  I understand that for debugging in development printing stack traces is handy, but in production crashing with a stack trace on an expected signal is messy.

I'll have a look at being a bit more specific with the exception handling.

Reply:
The KeyboardInterrupt stack trace was helpful in understanding issue #36; however, I believe the fix for that addressed the (very) intermittent hang that I had been chasing for months, so I think its usefulness has passed. And I agree that it's messy. I guess I should catch it in the other script, too.

Reply:
OK, since everything else has been resolved, I'm going to go ahead and accept this as-is and then change it to only catch KeyboardInterrupt (and Terminated) when I change the other scripts to do the same.